### PR TITLE
FEAT: support falcon-instruct 7b and 40b pytorch model

### PR DESCRIPTION
**Pull request by jiayini1119**

### CODE REVIEW

File: xinference/model/llm/llm_family.json [hunk 1]
```diff
@@ -617,5 +617,67 @@
...
+      }
+    ],
+    "prompt_style": {
+      "style_name": "FALCON",
+      "roles": [
+        "User",
+        "Assistant"
+      ],
+      "intra_message_sep": "\n"
+    }
```
Comment (pangyoki) [inline xinference/model/llm/llm_family.json:681]: - prompt_style
```
"prompt_style": {
      "style_name": "FALCON",
      "system_prompt": "",
      "roles": [
        "User",
        "Assistant"
      ],
      "intra_message_sep": "\n",
      "inter_message_sep": "<|endoftext|>",
      "stop": ["\nUser"],
      "stop_token_ids": [
        0,
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11
      ]
    }
```
- `FALCON` style_name:
```
        elif prompt_style.style_name == "FALCON":
            ret = prompt_style.system_prompt
            for message in chat_history:
                role = message["role"]
                content = message["content"]
                if content:
                    ret += (
                            role
                            + ": "
                            + content.replace("\r\n", "\n").replace("\n\n", "\n")
                    )
                    ret += "\n\n"
                else:
                    ret += role + ":"
            return ret
```
- add unittest for `FALCON` style_name

File: xinference/model/llm/pytorch/core.py [hunk 2]
```diff
@@ -215,7 +216,10 @@ def generate(
         def generator_wrapper(
             prompt: str, device: str, generate_config: PytorchGenerateConfig
         ) -> Iterator[CompletionChunk]:
-            if "falcon" in self.model_family.model_name:
```
Comment (pangyoki) [inline xinference/model/llm/pytorch/core.py:218]: `"falcon" in self.model_family.model_name:` already include `falcon` and `falcon-instruct`.
No need to change this command.

File: xinference/model/llm/pytorch/core.py [hunk 3]
```diff
@@ -241,7 +245,10 @@ def generator_wrapper(
         else:
             device = self._pytorch_model_config.get("device", "cuda")
         if not stream:
-            if "falcon" in self.model_family.model_name:
+            if (
```
Comment (pangyoki) [inline xinference/model/llm/pytorch/core.py:248]: "falcon" in self.model_family.model_name: already include falcon and falcon-instruct.
No need to change this command.

File: xinference/model/llm/llm_family.json [hunk 4]
```diff
@@ -617,5 +617,67 @@
       }
     ],
     "prompt_style": null
+  },
```
Comment (pangyoki) [inline xinference/model/llm/llm_family.json:620]: Can you help to add this config to `falcon` model_family？
```
{
        "model_format": "pytorch",
        "model_size_in_billions": 7,
        "quantizations": [
          "4-bit",
          "8-bit",
          "none"
        ],
        "model_id": "tiiuae/falcon-7b"
      },
```

File: xinference/model/llm/llm_family.json [hunk 5]
```diff
@@ -617,5 +617,125 @@
...
+        8,
+        9,
+        10,
+        11
+      ]
+    }
+  },
+  {
+    "version": 1,
+    "model_name": "falcon-instruct",
```
Comment (pangyoki) [inline xinference/model/llm/llm_family.json:695]: only need one `falcon-instruct` model_family.
`model_specs` contains 7b and 40b configs.
you can refer to other model_family like `vicuna`

File: xinference/model/llm/llm_family.json [hunk 6]
```diff
@@ -617,5 +617,125 @@
       }
     ],
     "prompt_style": null
+  },
+  {
+    "version": 1,
+    "model_name": "falcon",
```
Comment (pangyoki) [inline xinference/model/llm/llm_family.json:623]: only need one `falcon` model_family.
`model_specs` contains 7b and 40b configs.
you can refer to other model_family like `vicuna`

**Review by pangyoki — APPROVED** [review]
LGTM

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**